driftbase/api/friendships.py

```diff
@@ -165,35 +164,43 @@ def delete(self, friendship_id):
 
 @bp.route('/invites', endpoint='invites')
 class FriendInvitesAPI(MethodView):
+    class CreateInviteSchema(ma.Schema):
+        player_id = ma.fields.Integer(required=False, metadata=dict(description="The receiving player of the invite. Optional."))
+        expiration_time_seconds = ma.fields.Integer(required=False, metadata=dict(description="The expiration time of the invite in seconds."))
+        token_format = ma.fields.String(required=False, metadata=dict(description="The format of the token. Supported values: 'uuid' and 'wordlist'."))
+        worldlist_number_of_words = ma.fields.Integer(required=False, metadata=dict(description="How many words the token should be if the token format is 'wordlist'"))
 
     @bp.response(http_client.OK, InviteSchema(many=True))
     def get(self):
         """ List invites sent by current player """
         CorePlayer2 = aliased(CorePlayer)
         return g.db.query(FriendInvite, CorePlayer.player_name, CorePlayer2.player_name). \
-            join(CorePlayer, CorePlayer.player_id == FriendInvite.issued_to_player_id). \
+            join(CorePlayer, CorePlayer.player_id == FriendInvite.issued_to_player_id, isouter=True). \
             join(CorePlayer2, CorePlayer2.player_id == FriendInvite.issued_by_player_id). \
             filter(FriendInvite.issued_by_player_id == int(current_user["player_id"]),
                    FriendInvite.expiry_date > datetime.datetime.utcnow(),
                    FriendInvite.deleted.is_(False))
 
-    def post(self):
+    @bp.arguments(CreateInviteSchema)
+    def post(self, args):
         """
         New Friend token
         """
         sending_player_id = int(current_user["player_id"])
         try:
-            receiving_player_id = int(request.args.get("player_id"))
+            receiving_player_id = int(args.get("player_id"))
             self._validate_friend_request(receiving_player_id)
         except TypeError:
             receiving_player_id = None
 
-        token = str(uuid.uuid4())
-        expires_seconds = DEFAULT_INVITE_EXPIRATION_TIME_SECONDS
-        config = g.conf.tenant.get('friends')
-        if config:
-            expires_seconds = config['invite_expiration_seconds']
-        expires_seconds = expires_seconds
+        token_format = args.get("token_format") or _get_tenant_config_value("invite_token_format")
+        if token_format == "uuid":
+            token = str(uuid.uuid4())
+        elif token_format == "wordlist":
+            number_of_words = args.get("worldlist_number_of_words") or _get_tenant_config_value("invite_token_worldlist_number_of_words")
+            token = wordlist.get_word_combination(number_of_words)
+
+        expires_seconds = args.get("expiration_time_seconds") or _get_tenant_config_value("invite_expiration_seconds")
```

Comment:
What would be the arguments against having an upper limit here? A week or a month, say?

A lower limit is a bit harder to set, it might be very low if automation is somehow involved in replying, but perhaps a ten minute lower boundary would be a good start anyway. Suggestions?

Reply:
I'm indifferent about having an upper limit, but I feel like that is unnecessary.

Small tangent: 
I've had a pending friend request on Epic Game Store for about 2 years now.

For some reason, their decline friend request doesn't actually decline it, so I just ignore it and now realize that their expiry is either non existent or really high.

Reply:
Not having an upper limit increases the need for uniqueness IMO. Games are likely to have a sane default anyway.

You probably found a bug in EGS. :)

Reply:
On EGS you can't specify a limit actually, but it should work to reject.